pulp_smash/tests/platform/api_v2/test_repository.py

```diff
@@ -0,0 +1,226 @@
+                    )
+                else:
+                    self.assertNotIn('Location', response.headers)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete the created repositories."""
+        for response in cls.responses:
+            if response.status_code == 201:
+                requests.delete(
+                    cls.cfg.base_url + response.json()['_href'],
+                    **cls.cfg.get_requests_kwargs()
+                ).raise_for_status()
+
+
+class ReadUpdateDeleteSuccessTestCase(TestCase):
+    """Establish that we can read, update, and delete repositories.
+
+    This test assumes that the assertions in :class:`CreateSuccessTestCase` are
+    valid.
+
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create three repositories to read, update, and delete."""
+        cls.cfg = get_config()
+        cls.update_body = {
+            'delta': {
+                'display_name': rand_str(),
+                'description': rand_str()
+            }
+        }
+        cls.bodies = [{'id': rand_str()} for _ in range(3)]
+        cls.paths = []
+        for body in cls.bodies:
+            response = requests.post(
+                cls.cfg.base_url + REPOSITORY_PATH,
+                json=body,
+                **cls.cfg.get_requests_kwargs()
+            )
+            response.raise_for_status()
+            cls.paths.append(response.json()['_href'])
+
+        # Read, update, and delete the three repositories, respectively.
+        cls.read_response = requests.get(
+            cls.cfg.base_url + cls.paths[0],
+            **cls.cfg.get_requests_kwargs()
+        )
+        cls.update_response = requests.put(
+            cls.cfg.base_url + cls.paths[1],
+            json=cls.update_body,
+            **cls.cfg.get_requests_kwargs()
+        )
+        cls.delete_response = requests.delete(
+            cls.cfg.base_url + cls.paths[2],
+            **cls.cfg.get_requests_kwargs()
+        )
+
+    def test_status_code(self):
+        """Assert that each response has a 200 status code."""
+        expected_status_codes = zip(
+            ('read_response', 'update_response', 'delete_response'),
+            (200, 200, 202)
+        )
+        for attr, expected_status in expected_status_codes:
+            with self.subTest(attr):
+                self.assertEqual(
+                    getattr(self, attr).status_code,
+                    expected_status
+                )
+
+    def test_read_attributes(self):
+        """Assert that the read repository has the correct attributes."""
+        attributes = self.read_response.json()
+        self.assertLessEqual(
+            set(self.bodies[0].keys()),
+            set(attributes.keys())
+        )
+        attributes = {key: attributes[key] for key in self.bodies[0].keys()}
+        self.assertEqual(self.bodies[0], attributes)
+
+    def test_update_attributes(self):
+        """Assert that the repository has updated attributes."""
```

Comment:
Try splitting this test method into multiple methods. For example:

```python
def test_update_attributes_v1(self):
    """Assert that the update request did not spawn any tasks."""
    # Assert that `spawned_tasks` is present and sane.

def test_update_attributes_v2(self):
    """Assert that the updated repository has correct attributes."""
    # Assert that `result` is present and sane.
```

It's generally undesirable to have more than one assertion in a test case, as an assertion failure prevents you from running following assertions. It only really makes sense to place multiple assertions in a test case when one assertion would always trigger a failure in a second test case (in which case it's worth trying to make multiple test cases).